File: t/t_coverage_example.pl
#!/usr/bin/perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003 by Wilson Snyder. This program is free software; you can
# redistribute it and/or modify it under the terms of either the GNU
# General Public License or the Perl Artistic License.

use File::Spec;

scenarios(vlt => 1);

setenv('VERILATOR', "$ENV{VERILATOR_ROOT}/bin/verilator");

run(cmd => ["make -j4 -C submodules/example-systemverilog"],
    logfile => "$Self->{obj_dir}/sim.log",
    );

file_grep("$Self->{obj_dir}/sim.log", qr/.*\n-- DONE --/is);

ok(1);
1;

File: t/t_coverage_example.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('vlt')

test.setenv('VERILATOR', os.environ['VERILATOR_ROOT'] + "/bin/verilator")

test.run(cmd=["make -j4 -C submodules/example-systemverilog"],
         logfile=test.obj_dir + "/sim.log")

test.file_grep(test.obj_dir + "/sim.log", r'.*\n-- DONE --')

test.passes()
